seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/sink/RedisSinkWriter.java

```diff
@@ -60,23 +68,94 @@ public RedisSinkWriter(SeaTunnelRowType seaTunnelRowType, RedisParameters redisP
 
     @Override
     public void write(SeaTunnelRow element) throws IOException {
-        String data = new String(serializationSchema.serialize(element));
-        String keyField = redisParameters.getKeyField();
         List<String> fields = Arrays.asList(seaTunnelRowType.getFieldNames());
-        String key;
-        if (fields.contains(keyField)) {
-            key = element.getField(fields.indexOf(keyField)).toString();
-        } else {
-            key = keyField;
-        }
+        String key = getKey(element, fields);
         keyBuffer.add(key);
-        valueBuffer.add(data);
+        String value = getValue(element, fields);
+        valueBuffer.add(value);
         if (keyBuffer.size() >= batchSize) {
             doBatchWrite();
             clearBuffer();
         }
     }
 
+    private String getKey(SeaTunnelRow element, List<String> fields) {
+        String keyField = redisParameters.getKeyField();
+        String[] keyFieldSegments = keyField.split(REDIS_GROUP_DELIMITER);
+        StringBuilder key = new StringBuilder();
+        for (int i = 0; i < keyFieldSegments.length; i++) {
+            String keyFieldSegment = keyFieldSegments[i];
+            if (keyFieldSegment.startsWith(LEFT_PLACEHOLDER_MARKER)
+                    && keyFieldSegment.endsWith(RIGHT_PLACEHOLDER_MARKER)) {
+                String realKeyField = keyFieldSegment.substring(1, keyFieldSegment.length() - 1);
+                if (fields.contains(realKeyField)) {
+                    key.append(element.getField(fields.indexOf(realKeyField)).toString());
+                } else {
+                    key.append(keyFieldSegment);
+                }
+            } else {
+                key.append(keyFieldSegment);
+            }
+            if (i != keyFieldSegments.length - 1) {
+                key.append(REDIS_GROUP_DELIMITER);
+            }
+        }
+        return key.toString();
+    }
+
+    private String getValue(SeaTunnelRow element, List<String> fields) {
+        String value;
+        RedisDataType redisDataType = redisParameters.getRedisDataType();
+        if (RedisDataType.HASH.equals(redisDataType)) {
+            value = handleHashType(element, fields);
+        } else {
+            value = handleOtherTypes(element, fields);
+        }
+        if (StringUtils.isEmpty(value)) {
+            byte[] serialize = serializationSchema.serialize(element);
+            value = new String(serialize);
+        }
```

Comment:
why not just put "" when value equals ""?

Reply:
If value is ", it means that the user has not set a custom value or a custom hash key and hash value, then value is the entire json data

Reply:
got it.

Reply:
If the value of the data is originally "", the current logic will cause data abnormality. Do I understand it correctly?

Reply:
I understand that upstream data should at least have field names

Reply:
For example:
|id | name | age|
| --- | --- | --- |
|1| | 22|

The name is "".
The redis sink config is 
```
{
 key = {id}
 value_field = name
 data_type = key
}
```
The value should be `1 -> `, but it would be `1 -> {"id":1,"name":"","age":22}`

Reply:
got it. I will fix and update the PR.